### PR TITLE
fix(rust): make sure all the handshake spans end up in the same trace with the secure channel use

### PR DESCRIPTION
**Pull request by etorreborre**
This PR makes sure that:

 - The `trace_code` function sets-up the initial `Context` with the root tracing context.
 - When a `Context` is duplicated, its tracing context is updated with the most recent span.

This way, the spans corresponding to starting the secure channel end up under the same root span as the code using the secure channel
```
root
└── send_echo_message_over_secure_channel
    ├── create tcp transport
    ├── create tcp transport
    ├── TcpSendWorker::handle_message
    |   └── TcpRecvProcessor::process
    |       └── HandshakeWorker::handle_message
    |           └── TcpSendWorker::handle_message
    |               └── TcpRecvProcessor::process
    |                   └── HandshakeWorker::handle_message
    |                       └── TcpSendWorker::handle_message
    |                           └── TcpRecvProcessor::process
    |                               └── HandshakeWorker::handle_message
    └── MessageSender::handle_message
        └── EncryptorWorker::handle_message
            └── handle_encrypt
                └── TcpSendWorker::handle_message
                    └── TcpRecvProcessor::process
                        └── DecryptorWorker::handle_message
                            └── handle_decrypt
                                └── Echoer::handle_message
                                    └── EncryptorWorker::handle_message
                                        └── handle_encrypt
                                            └── TcpSendWorker::handle_message
                                                └── TcpRecvProcessor::process
                                                    └── DecryptorWorker::handle_message
                                                        └── handle_decrypt
```

**UPDATE**: I've added a fix for the creation of a user journey. I don't know but there was a regression there. 
**UPDATE**: I've added another fix the selection of crates when logging is off.

### CODE REVIEW

File: implementations/rust/ockam/ockam_api/src/logs/logging_configuration.rs [hunk 1]
```diff
@@ -151,18 +167,19 @@ impl LoggingConfiguration {
     }
 
     /// Return a LoggingConfiguration which doesn't log anything
-    pub fn off() -> LoggingConfiguration {
-        LoggingConfiguration::new(
+    /// We still retrieve the
```
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_api/src/logs/logging_configuration.rs]: Some text is missing here?

File: implementations/rust/ockam/ockam_node/src/context/context_lifecycle.rs [hunk 2]
```diff
@@ -119,7 +119,7 @@ impl Context {
             self.transports.clone(),
             &self.flow_controls,
             #[cfg(feature = "std")]
-            self.tracing_context(),
+            OpenTelemetryContext::current(),
```
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_node/src/context/context_lifecycle.rs:122]: What's the rationale behind this change? I'm trying to understand when we need to get the "global" context or just copy the one available at `self`.

File: implementations/rust/ockam/ockam_api/src/logs/logging_configuration.rs (hunk 1 above)
Comment (etorreborre) [reply implementations/rust/ockam/ockam_api/src/logs/logging_configuration.rs]: Oops, I must have be interrupted. I forced-push the rest of the comment.
But I am thinking that I will probably refactor that part to extract the configuration that's used by both logging and tracing to make things clearer.

File: implementations/rust/ockam/ockam_node/src/context/context_lifecycle.rs (hunk 2 above)
Comment (etorreborre) [reply implementations/rust/ockam/ockam_node/src/context/context_lifecycle.rs:122]: That's all due to the complexity of:

 - Having to pass the current tracing context between async processes when working with our workers. We work around this by setting a tracing context on the worker Context
 
 - Having a different notion of "tracing context" between the tracing crate, which creates spans using the `#[instrument]` macro, and the opentelemetry crate which creates the span data sent to the collector at a different point in time (when exiting the instrumented function). It's also the `opentelemetry` crate which generates span ids. So in order to get the latest span id, the one corresponding to the span created by the latest `#[instrument]`ed method, we need to reactualize the OpenTelemetry context using all the code in `OpenTelemetryContext::current()`.

**Review by SanjoDeundiak — APPROVED** [review]
